src/hooks/useTodoContext.tsx

```diff
@@ -0,0 +1,53 @@
+// https://github.com/CEOS-Developers/react-todo-15th/pull/11/files#diff-4fd5b35d6c3fbfd63c97f711bd5fe3408b11551836f6cec1ecfb67a498d2843a
+// custom hook에 관한 부분 김선종 멘트님 코드 많이 참고했습니다..!
+
+import { useContext } from 'react';
+import { TodoContext } from '../context/todoProvider';
+import { v4 as uuidv4 } from 'uuid';
+import { runToast } from '../components/toast/Toast';
+import { ITodoItem } from '../Interface';
+
+const useTodoContext = () => {
+    const { list, setList } = useContext(TodoContext);
+
+    const addItem = (value: string): void => {
+        const createItem = () => {
+            setList([
+                ...list,
+                {
+                    id: uuidv4(),
+                    content: value,
+                    done: false,
+                },
+            ]);
+        };
+
+        if (value) {
+            if (list.find((item) => item.content === value)) {
+                runToast('동일한 내용으론 작성할 수 없어요!');
+            } else {
+                createItem();
+            }
```

Comment:
제가 기억하기로는 1주차 코드에서는 content로 delete 연산을 진행했기 때문에 동일한 내용을 작성할 수 없도록 처리한 걸로 아는데요 이번에는 id로 delete 연산을 진행하니까 동일한 내용 필터링하는 부분은 이제 없어도 되지 않을까 하는 개인적인 생각입니다..!
```suggestion
        if (value.replace(/\s/g, "")) {
            createItem();
```
그리고 value 자체만 검증하면 공백만 입력된 경우에도 할 일로 추가되기 때문에 정규식 사용해서 공백 제거한 후 검증하는 코드로 작성해봐도 괜찮을 것 같아요!

Reply:
꼼꼼한 리뷰 감사드립니다..!!